### PR TITLE
add scaling

### LINKED ISSUE #20: Scaling for chart with numeric axis.
When much data conveniently have opportunity for scaling. Guess it applicable for all chart with numeric axis.

### PR DESCRIPTION
**Pull request by Kvaz1r**
Closed #20 . Also changed tabs for spaces as in majority files. 
I add new method for base chart instead create new, because chart type where such opportunity not need just not will call corresponding method. 

But here still exist some things which I want discuss. I will write it as comments.

### CODE REVIEW

File: include/wx/charts/wxchartgrid.h [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
-	all copies or substantial portions of the Software.
-
-	THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-	IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-	FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
-	THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-	LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
-	FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
-	IN THE SOFTWARE.
+    Copyright (c) 2016-2017 Xavier Leclercq and the wxCharts contributors.
```
Comment (Kvaz1r) [inline include/wx/charts/wxchartgrid.h]: I guess that this form would be more conviently than list of names.

File: include/wx/charts/wxchartgrid.h [hunk 2]
```diff
@@ -102,49 +102,67 @@ class wxChartGrid : public wxChartElement
...
 private:
-	wxChartGridOptions m_options;
-	wxPoint2DDouble m_position;
-	wxChartAxis::ptr m_XAxis;
-	wxChartAxis::ptr m_YAxis;
-	wxChartGridMapping m_mapping;
-	// Whether something has changed and we
-	// need to rearrange the chart
-	bool m_needsFit;
+    struct Corners
```
Comment (Kvaz1r) [inline include/wx/charts/wxchartgrid.h]: Store value of corner points for chart. Maybe there is better name for the struct.

File: include/wx/charts/wxmath2dplotctrl.h [hunk 3]
```diff
@@ -72,6 +72,9 @@ class wxMath2DPlotCtrl : public wxChartCtrl
 private:
     wxMath2DPlot m_math2dPlot;
     wxMenu m_contextMenu;
+    wxMenu *m_subMenu;
```
Comment (Kvaz1r) [inline include/wx/charts/wxmath2dplotctrl.h]: If instead pointer used just wxMenu member - got an runtime error for unknown reasons to me.

File: include/wx/charts/wxchartgrid.h (hunk 1 above)
Comment (xleclercq) [inline include/wx/charts/wxchartgrid.h]: I read a bit about the way to manage this and what you suggest seems to be what most projects do.  I will also add an AUTHORS file. Regardless of the copyright notice though you own the copyright of the changes you made to the files. The copyright notice just makes it explicit.

File: include/wx/charts/wxmath2dplotctrl.h (hunk 3 above)
Comment (xleclercq) [inline include/wx/charts/wxmath2dplotctrl.h]: You use AppendSubMenu to add the menu to m_contextMenu so m_contextMenu takes ownership. So it's correct it needs to be a pointer and m_contextMenu will delete it when it is no longer needed.

File: include/wx/charts/wxchartgrid.h (hunk 2 above)
Comment (xleclercq) [inline include/wx/charts/wxchartgrid.h]: Yes Corners isn't a very good name. But it's only internal to the class so it's not too important. What about AxisLimits?

Comment (Kvaz1r) [inline include/wx/charts/wxchartgrid.h]: Exactly what need. I'll update it. What about name of variables 
```
 m_curcorners;// short form for current corners
 m_orcorners;// original
```
change to 
```
m_CurAxisLimits;
m_OrAxisLimits;
```
or better use something else?

Comment (xleclercq) [inline include/wx/charts/wxchartgrid.h]: I would call them m_curAxisLimits and m_origAxisLimits.

Comment (Kvaz1r) [inline include/wx/charts/wxchartgrid.h]: Good, I will update it. Something else needs to be changed?

**Comment by Kvaz1r** [comment]
updated with the mentioned above comments.

**Pull request merged**